MH4U Dex/HomeViewController.m

```diff
@@ -28,6 +30,21 @@ - (IBAction)resetButtonPressed:(id)sender
     [CoreDataController setShouldTriggerReloadUponRestart:YES];
 }
 
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    [MBProgressHUD showHUDAddedTo:self.view animated:YES];
+    dispatch_queue_t queue = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
+    dispatch_async(queue, ^{
+        [[CoreDataController sharedCDController] tryLoadSequence];
```

Comment:
I think the best mechanism would be to call `tryLoadSequence` on the main thread and have it dispatch to the background if it's loading from JSON.

To do the loading from JSON, create a separate `NSManagedObject` instance on the background thread, using the same persistent store; do all the loading with that managed object context; save it; then switch back to the main thread and make the shared context sync up with the persistent store.